src/olympia/reviewers/models.py

```diff
@@ -798,6 +801,52 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @classmethod
+    def get_base_addons(cls):
+        """Return base queryset of add-ons we consider when we look at growth
+        thresholds.
+
+        This is a classmethod, it is not specific to a particular tier."""
+        return Addon.unfiltered.exclude(status=amo.STATUS_DISABLED).filter(
+            type=amo.ADDON_EXTENSION
+        )
+
+    def get_tier_boundaries(self):
+        """Return boundaries used to filter add-ons for that tier instance."""
+        return {
+            'average_daily_users__gte': self.lower_adu_threshold,
+            'average_daily_users__lt': self.upper_adu_threshold,
+        }
+
+    @cached_property
+    def average_growth(self):
+        return (
+            self.get_base_addons()
+            .filter(**self.get_tier_boundaries())
+            .aggregate(Avg('hotness', default=0))
+            .get('hotness__avg')
+        )
+
+    def get_growth_threshold(self):
+        """Return the growth threshold for that tier, or None.
+
+        The value is computed from the average growth (hotness) of add-ons in
+        that tier plus the growth_threshold_before_flagging converted to
+        decimal."""
+        return self.average_growth + self.growth_threshold_before_flagging / 100
```

Comment:
With the old algorithm this would have been:
```python
        return self.average_growth * (
            1 + self.growth_threshold_before_flagging / 100
        )